idaes/apps/grid_integration/multiperiod/design_and_operation_models.py

```diff
@@ -0,0 +1,138 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+from pyomo.environ import (
+    Var,
+    Param,
+    Binary,
+    Expression,
+    NonNegativeReals,
+    Constraint,
+)
+from functools import reduce
+from idaes.core.base.process_base import declare_process_block_class
+from idaes.models.unit_models import SkeletonUnitModelData
+from pyomo.common.config import ConfigValue, In
+
+
+def deepgetattr(obj, attr):
```

Comment:
@lbianchi-lbl I borrowed this function from internet. Is it okay to have it in the repository? If not, is there an alternative for this function?

Reply:
@radhakrishnatg the Python standard library has [operator.attrgetter()](https://docs.python.org/3/library/operator.html#operator.attrgetter) that you can use instead:

```
import operator

m.fs.unit = MyUnitModel()

assert m.fs.unit is operator.attrgetter("fs.unit")(m)
```

Reply:
@radhakrishnatg What exactly are you trying to test with this anyway? I suspect there are better ways to do whatever you are trying to do using native Pyomo functionality.

Also, on borrowing from the internet, it depends on where you got it and what license they place on the code. E.g., you cannot copy code from StackOverflow into IDAES as the licenses are incompatible.

Reply:
Thank you @lbianchi-lbl and @andrewlee94 ! 
@andrewlee94 I want a method similar to `getattr` which also works for attribute of an attribute e.g., `modified_getattr(m, "fs.compressor[1].input")`. Ludovico's suggestion works for me. Please let me know if there are better methods available in Pyomo.

Reply:
I'm aware of Pyomo's [find_component()](https://pyomo.readthedocs.io/en/stable/library_reference/aml/index.html#pyomo.environ.ConcreteModel.find_component) method, but the Pyomo folks will have the proper answer.

Reply:
@radhakrishnatg What is your actual use case for this? I would hesitate to write an entirely new function to do this, as there is likely things that will already do that for you. For one, as @lbianchi-lbl noted there is the `find_component()`.

Reply:
I've updated the files to use Pyomo's native `find_component()` function instead of the borrowed `deepgetattr()` function.